src/NexusMods.DataModel/Games/GameLocationsRegister.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System.Diagnostics;
+using NexusMods.Paths;
+
+namespace NexusMods.DataModel.Games;
+
+/// <summary>
+/// Installation specific register of game locations <see cref="GameFolderType"/> and their resolved paths.
+/// <remarks>
+/// This class provides information about the locations, such as resolved paths, nested locations and top level parents.
+/// </remarks>
+/// </summary>
+public class GameLocationsRegister
+{
+    private readonly Dictionary<GameFolderType, GameLocationDescriptor> _locations = new();
+
+    /// <summary>
+    /// Obtain the resolved path for a <see cref="GameFolderType"/>.
+    /// </summary>
+    /// <param name="id">The <see cref="GameFolderType"/> to lookup</param>
+    public AbsolutePath this[GameFolderType id] => _locations[id].ResolvedPath;
+
+    /// <summary>
+    /// Construct a new instance of <see cref="GameLocationsRegister"/>.
+    /// </summary>
+    /// <remarks>
+    /// Computes the nested and top level parents relationships.
+    /// </remarks>
+    /// <param name="newLocations">A Dictionary of <see cref="GameFolderType"/>, <see cref="AbsolutePath"/> to register</param>
+    public GameLocationsRegister(IReadOnlyDictionary<GameFolderType, AbsolutePath> newLocations)
+    {
+        foreach (var (newId, newPath) in newLocations)
+        {
+            var newLocation = new GameLocationDescriptor(newId, newPath);
+
+            var isDuplicate = !_locations.TryAdd(newId, newLocation);
```

Comment:
Inline this method call into the assert, because chances are this will still execute in Release mode; the dead code eliminator isn't super sophisticated and might not consider this pure enough to remove.

Reply:
I always want to add, I just check whether it failed due to duplicated ID in debug mode. Unless I'm getting your comment wrong.

Reply:
Ah, okay. In that case that's no problem. I assumed it was just for debug since the variable went unused in Release; all good.